packages/medusa/src/api-v2/admin/api-keys/[id]/sales-channels/batch/route.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  addSalesChannelsToApiKeyWorkflow,
+  removeSalesChannelsFromApiKeyWorkflow,
+} from "@medusajs/core-flows"
+import { ApiKeyType, MedusaError } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../../types/routing"
+import { AdminApiKeySalesChannelType } from "../../../validators"
+import { BatchMethodRequest } from "@medusajs/types"
+import { refetchApiKey } from "../../../helpers"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<
+    BatchMethodRequest<AdminApiKeySalesChannelType, AdminApiKeySalesChannelType>
+  >,
+  res: MedusaResponse
+) => {
+  const { create, delete: toDelete } = req.validatedBody
+  const apiKey = await refetchApiKey(
+    req.params.id,
+    req.scope,
+    req.remoteQueryConfig.fields
+  )
+
+  if (apiKey.type !== ApiKeyType.PUBLISHABLE) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      "Sales channels can only be associated with publishable API keys"
+    )
+  }
+
+  if (create && create.length) {
+    const workflowInput = {
+      data: [
+        {
+          api_key_id: req.params.id,
+          sales_channel_ids: create ?? [],
+        },
+      ],
+    }
+
+    const { errors } = await addSalesChannelsToApiKeyWorkflow(req.scope).run({
+      input: workflowInput,
+      throwOnError: false,
+    })
+
+    if (Array.isArray(errors) && errors[0]) {
+      throw errors[0].error
+    }
+  }
+
+  if (toDelete && toDelete.length) {
+    const workflowInput = {
+      data: [
+        {
+          api_key_id: req.params.id,
+          sales_channel_ids: toDelete,
+        },
+      ],
+    }
+
+    const { errors } = await removeSalesChannelsFromApiKeyWorkflow(
```

Comment:
q: any reason why these aren't in one workflow?

Context - if the delete fails, you can't revert the creates. If the case is that only one is running at a time, maybe we can do an XOR validation with zod

Reply:
@riqwan I'll move them to a single workflow, it was just too many changes to do at once so I figured I'll just make the user-facing HTTP interface to what we want it to be first